lib/src/cpp/utils/ParseXML.cpp

```diff
@@ -260,25 +264,53 @@ xercesc::DOMNode* ParseXML::get_node(
                                       tag_name + " does not have the attribute " + att_name + "\n");
                         }
                     }
-                    // LIST element TODO refactor
-                    else if (&index == nullptr)
+                    // LIST element
+                    else if (index != nullptr)
                     {
-                        int32_t realIndex = index;
-                        // Transform index to real index
-                        if (index < 0)
+                        // Check value of index
+                        std::string in = *index;
+
+                        // Empty index
+                        if (in == "")
                         {
-                            realIndex = node_tag_list->getLength() + index;
+                            // return parent
+                            return tag_node;
                         }
-                        // Check bounds
-                        if (realIndex < 0 || realIndex >= node_tag_list->getLength())
+                        // Index has a value
+                        else
                         {
-                            // Throw eprosima::qosprof::ElementNotFound exception
-                            throw ElementNotFound(
-                                      tag_name + " does not have an element in position " + std::to_string(
-                                          realIndex) + "\n");
+                            // Parse index value and obtain node
+                            int32_t int_index = 0;
+                            int32_t real_index = 0;
```

Comment:
I think the same variable can be used because there is no need to keep both values at the same time.